src/converter.ts

```diff
@@ -385,13 +406,39 @@ export class Converter {
 
     const files: File[] = []
 
-    await this.usePuppeteer(html, async (page, { render }) => {
-      await page.setViewport({
-        ...tpl.rendered.size,
-        deviceScaleFactor: opts.scale ?? 1,
-      })
+    await this.usePuppeteer(html, async (page, { browser, render }) => {
+      const scale = opts.scale ?? 1
+      const viewPort = (
+        browser.kind === 'firefox'
+          ? ({
+              width: Math.floor(tpl.rendered.size.width * scale),
+              height: Math.floor(tpl.rendered.size.height * scale),
+              // Firefox seems not to respect deviceScaleFactor when taking screenshot
+              // https://github.com/puppeteer/puppeteer/issues/13032
+              // https://github.com/w3c/webdriver-bidi/issues/686
+              deviceScaleFactor: 1,
+            } as const)
+          : ({
+              width: tpl.rendered.size.width,
+              height: tpl.rendered.size.height,
+              deviceScaleFactor: scale,
+            } as const)
+      ) satisfies Viewport
+
+      await page.setViewport(viewPort)
       await render()
-      await page.emulateMediaType('print')
+      await page.addStyleTag({
+        content: ':root,body { scrollbar-width:none !important; }',
+      })
+
+      try {
+        await page.emulateMediaType('print')
```

Comment:
WebDriver BiDi does not yet support media type emulation, but it's OK because using `bare` template, that is almost compatible with the print media, whenever converting to image(s).